src/pages/Dashboard/MyAccount/MyAccountUi.tsx

```diff
@@ -0,0 +1,217 @@
+  netApyPercentage: number | undefined;
+  dailyEarningsCents: number | undefined;
+  supplyBalanceCents: number | undefined;
+  borrowBalanceCents: number | undefined;
+  borrowLimitCents: number | undefined;
+  safeBorrowLimitPercentage: number;
+  withXvs: boolean;
+  onXvsToggle: (newValue: boolean) => void;
+  className?: string;
+}
+
+export const MyAccountUi = ({
+  netApyPercentage,
+  dailyEarningsCents,
+  supplyBalanceCents,
+  borrowBalanceCents,
+  borrowLimitCents,
+  safeBorrowLimitPercentage,
+  withXvs,
+  onXvsToggle,
+  className,
+}: IMyAccountUiProps) => {
+  const styles = useStyles();
+  const { t, Trans } = useTranslation();
+
+  const handleXvsToggleChange: IToggleProps['onChange'] = (_event, checked) => onXvsToggle(checked);
+
+  const readableBorrowBalance =
+    typeof borrowBalanceCents === 'number'
+      ? formatCentsToReadableValue(borrowBalanceCents)
+      : undefined;
+
+  let borrowLimitUsedPercentage: number | undefined;
+  if (borrowLimitCents === 0) {
+    borrowLimitUsedPercentage = 0;
+  } else if (typeof borrowBalanceCents === 'number' && typeof borrowLimitCents === 'number') {
+    borrowLimitUsedPercentage = Math.round((borrowBalanceCents * 100) / borrowLimitCents);
+  }
+
+  const readableBorrowLimitUsedPercentage =
+    typeof borrowLimitUsedPercentage === 'number' ? `${borrowLimitUsedPercentage}%` : undefined;
+
+  const safeBorrowLimitCents =
+    typeof borrowLimitCents === 'number'
+      ? Math.floor((borrowLimitCents * safeBorrowLimitPercentage) / 100)
+      : undefined;
+
+  const readableSafeBorrowLimit =
+    typeof safeBorrowLimitCents === 'number'
+      ? formatCentsToReadableValue(safeBorrowLimitCents, true)
+      : undefined;
+
+  return (
+    <div css={styles.container} className={className}>
+      <div css={[styles.row, styles.header]}>
+        <Typography variant="h4">{t('myAccount.title')}</Typography>
+
+        <Typography component="div" variant="small2" css={styles.apyWithXvs}>
+          <Tooltip css={styles.tooltip} title={t('myAccount.apyWithXvsTooltip')}>
+            <Icon css={styles.infoIcon} name="info" />
+          </Tooltip>
+
+          <Typography color="text.primary" variant="small1" css={styles.apyWithXvsLabel}>
+            {t('myAccount.apyWithXvs')}
+          </Typography>
+
+          <Toggle css={styles.toggle} value={withXvs} onChange={handleXvsToggleChange} />
+        </Typography>
+      </div>
+
+      <div css={styles.netApyContainer}>
+        <div css={styles.netApy}>
+          <Typography component="div" variant="small2" css={styles.netApyLabel}>
+            {t('myAccount.netApy')}
+          </Typography>
+
+          <Tooltip css={styles.tooltip} title={t('myAccount.netApyTooltip')}>
+            <Icon css={styles.infoIcon} name="info" />
+          </Tooltip>
+        </div>
+
+        <Typography variant="h1" color="interactive.success">
+          {typeof netApyPercentage === 'number' ? `${netApyPercentage}%` : '-'}
```

Comment:
Should this be formatted from a function?

Reply:
This is the same duplicated logic where we check if a value is a number, if it is we do something, if it isn't we return a `-`